src/test/java/de/fraunhofer/aisec/cpg/enhancements/variableResolution/VariableResolverJavaTest.java

```diff
@@ -0,0 +1,305 @@
+package de.fraunhofer.aisec.cpg.enhancements.variableResolution;
+
+import de.fraunhofer.aisec.cpg.TranslationConfiguration;
+import de.fraunhofer.aisec.cpg.TranslationManager;
+import de.fraunhofer.aisec.cpg.graph.CallExpression;
+import de.fraunhofer.aisec.cpg.graph.DeclaredReferenceExpression;
+import de.fraunhofer.aisec.cpg.graph.Expression;
+import de.fraunhofer.aisec.cpg.graph.FieldDeclaration;
+import de.fraunhofer.aisec.cpg.graph.ForStatement;
+import de.fraunhofer.aisec.cpg.graph.Literal;
+import de.fraunhofer.aisec.cpg.graph.MemberExpression;
+import de.fraunhofer.aisec.cpg.graph.MethodDeclaration;
+import de.fraunhofer.aisec.cpg.graph.Node;
+import de.fraunhofer.aisec.cpg.graph.ParamVariableDeclaration;
+import de.fraunhofer.aisec.cpg.graph.RecordDeclaration;
+import de.fraunhofer.aisec.cpg.graph.TranslationUnitDeclaration;
+import de.fraunhofer.aisec.cpg.graph.ValueDeclaration;
+import de.fraunhofer.aisec.cpg.graph.VariableDeclaration;
+import de.fraunhofer.aisec.cpg.helpers.NodeComparator;
+import de.fraunhofer.aisec.cpg.helpers.SubgraphWalker;
+import de.fraunhofer.aisec.cpg.helpers.Util;
+import java.io.File;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+
+// Todo VariableResolver 7 Failed, 7 Passed
+
+@Disabled(
+    "Until changing variable resolution to ScopeManager. Then in detail disable the tests that need specific fixes")
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class VariableResolverJavaTest {
+
+  // Externally defined static global
+
+  private RecordDeclaration externalClass;
+  private FieldDeclaration externVarName;
+  private FieldDeclaration externStaticVarName;
+
+  private RecordDeclaration outerClass;
+  private FieldDeclaration outerVarName;
+  private FieldDeclaration outerStaticVarName;
+  private FieldDeclaration outerImpThis;
+
+  private RecordDeclaration innerClass;
+  private FieldDeclaration innerVarName;
+  private FieldDeclaration innerStaticVarName;
+  private FieldDeclaration innerImpThis;
+  private FieldDeclaration innerImpOuter;
+
+  private MethodDeclaration main;
+
+  private MethodDeclaration outer_function1;
+  private List<ForStatement> forStatements;
+  private MethodDeclaration outer_function2;
+  private MethodDeclaration outer_function3;
+  private MethodDeclaration outer_function4;
+
+  private MethodDeclaration inner_function1;
+  private MethodDeclaration inner_function2;
+  private MethodDeclaration inner_function3;
```

Comment:
Lot of copy/paste from the other test?!

Reply:
The class structure was intentionally kept similar, so it made sense to do so and then add only the variables structures that were not part of the other test class.